app/controllers/images_controller.rb

```diff
@@ -102,7 +102,7 @@ def update
   def revert
     @warpable = Warpable.find params[:id]
     version = @warpable.versions.find(params[:version])
-    version.reify&.save
+    version.reify
```

Comment:
This works good for all cases, except, we need to store a version when we first revert. So, do one thing -- Put if condition here something like `if count_version == 1`, then we are changing in-between reverts so do reify, `@warpable.count_version = 1` and save in this if block. Otherwise, reify.&save so that version would be created in case where we are reverting for first time. And, in update count_version = 0 is there so we would always create a version.

Reply:
On every update a new version is created and stored. And on revert, we are being taken to that version. So if I will use the if condition, a cloned copy of the reverted version would be added to the top of the stack, beacuse previously all the versions are already stored. I guess it would work fine without if. What do you think?

Reply:
When we revert for the first time, in that case without `reify.save`. That point from where we are reverting is getting lost that's why we want to save for that case.